src/TestableIO.System.IO.Abstractions/IFileSystemExtensibility.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Diagnostics.CodeAnalysis;
+
+namespace System.IO.Abstractions
+{
+    /// <summary>
+    ///     A container to give access to the wrapped instance and to store metadata on <see cref="IFileSystem" /> entities.
+    ///     <br />
+    ///     This is useful for writing extension methods.
+    /// </summary>
+    public interface IFileSystemExtensibility
```

Comment:
I don't fully understand why we need this. Can you give a specific example where this is useful?

Reply:
On .NET core, Microsoft implemented the ACL-Functionality using extension methods (see [FileSystemAclExtensions](https://learn.microsoft.com/en-us/dotnet/api/system.io.filesystemaclextensions?view=net-6.0).
When looking for a way to do something similar, I implemented this extensibility method, so I could
- Get the underlying instance if it is there and execute the extension method on it or
- Store/Retrieve some data on the mocked instance and work with this storage mechanism instead

After reading your comment, I think it would make more sense, if I remove it from this PR and add it in again later, together with the ACL-Refactoring, when the use case makes it clear what it is for...